core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -674,13 +596,9 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
    * elector
    */
   def startup() = {
-    inLock(controllerContext.controllerLock) {
-      info("Controller starting up")
-      registerSessionExpirationListener()
-      isRunning = true
-      controllerElector.startup
-      info("Controller startup complete")
-    }
+    controllerEventQueue.put(Startup)
+    controllerEventQueue.put(Elect)
```

Comment:
Possible, but I'd rather we try to stick to the following pattern if we can:
all actions that modify controller state should be done as a ControllerEvent processed by the ControllerThread.